src/ports/rt-kernel/dwmac1000.c

```diff
@@ -280,6 +297,68 @@ static void dwmac1000_pbuf_dump (struct pbuf * p)
 }
 #endif /* DEBUG_DATA */
 
+/**
+ * Calculate MAU type
+ *
+ * @param link_state       In:    Link state.
+ * @return The MAU type
+ */
+static pnal_eth_mau_t dwmac1000_mau_type (uint8_t link_state)
+{
+   switch (link_state)
+   {
+   case PHY_LINK_OK | PHY_LINK_10MBIT:
+      return PNAL_ETH_MAU_COPPER_10BaseT;
+   case PHY_LINK_OK | PHY_LINK_100MBIT:
+      return PNAL_ETH_MAU_COPPER_100BaseTX_HALF_DUPLEX;
+   case PHY_LINK_OK | PHY_LINK_1000MBIT:
+      return PNAL_ETH_MAU_COPPER_1000BaseT_HALF_DUPLEX;
+   case PHY_LINK_OK | PHY_LINK_10MBIT | PHY_LINK_FULL_DUPLEX:
+      return PNAL_ETH_MAU_COPPER_10BaseT;
+   case PHY_LINK_OK | PHY_LINK_100MBIT | PHY_LINK_FULL_DUPLEX:
+      return PNAL_ETH_MAU_COPPER_100BaseTX_FULL_DUPLEX;
+   case PHY_LINK_OK | PHY_LINK_1000MBIT | PHY_LINK_FULL_DUPLEX:
+      return PNAL_ETH_MAU_COPPER_1000BaseT_FULL_DUPLEX;
+   default:
+      return PNAL_ETH_MAU_RADIO;
+   }
+}
+
+/**
+ * Calculate advertised capabilites
+ *
+ * @param anar       In:    Contents of the ANAR register read from PHY.
+ * @return Profinet advertised capabilites as a bitfield.
+ */
+static uint16_t dwmac1000_advertised_capabilities (uint16_t anar)
```

Comment:
Conversion to PNAL types should preferrably happen in pnal.c. Ideally the ethernet driver should not need to know about profinet.

Reply:
Moved conversion to PNAL types to pnal.c.

Reply:
Done.